src/tooling/migration-rrule-recurrence/migration-rrule-recurrence.js

```diff
@@ -76,34 +76,29 @@ async function update(collectionName, documents) {
             bysetpos: [],
             bymonthday: []
         };
-        delete card.timeSpans;
 
-        delete card.data.hours;
-        delete card.data.minutes;
-        delete card.data.daysOfWeek;
-        delete card.data.months;
-        card.data.freq = rRule.freq;
-        card.data.byhour = rRule.byhour;
-        card.data.byminute = rRule.byminute;
-        card.data.byweekday = rRule.byweekday;
-        card.data.bymonth = rRule.bymonth;
-        card.data.bysetpos = rRule.bysetpos;
-        card.data.bymonthday = rRule.bymonthday;
+        const queryForDelete = {_id: card._id};
+        let replacementCard = card;
 
-        const queryForUpdate = {_id: card._id};
-        const newValuesForUpdate = {
-            $set: {
-                process: newProcess,
-                state: newState,
-                data: card.data,
-                rRule: rRule
-            },
-            $unset: {
-                timeSpans: ""
-            }
-        };
-        // execute update query
-        await db.collection(collectionName).updateOne(queryForUpdate, newValuesForUpdate);
+        delete replacementCard.timeSpans;
+        delete replacementCard.data.hours;
+        delete replacementCard.data.minutes;
+        delete replacementCard.data.daysOfWeek;
+        delete replacementCard.data.months;
+        replacementCard.data.freq = rRule.freq;
+        replacementCard.data.byhour = rRule.byhour;
+        replacementCard.data.byminute = rRule.byminute;
+        replacementCard.data.byweekday = rRule.byweekday;
+        replacementCard.data.bymonth = rRule.bymonth;
+        replacementCard.data.bysetpos = rRule.bysetpos;
+        replacementCard.data.bymonthday = rRule.bymonthday;
+        replacementCard._id = newProcess + '.' + card.processInstanceId;
```

Comment:
You need as well to add the processStateKey (to be set to processId.state)

Reply:
Done.